frontend/src/components/projects/ProjectsAssociatedSamples.tsx

```diff
@@ -40,22 +42,95 @@ const useLastProtocols = (sampleIDs: readonly Sample['id'][]) => {
+    const [orderByStepOrder, setOrderByStepOrder] = useState<Record<WorkflowStepOrder['id'], WorkflowStepOrder['order'] | undefined>>({})
+
+    const refresh = useCallback(async (sampleIDs: readonly Sample['id'][]) => {
+        const sampleNextStepByStudies = (await dispatch(api.sampleNextStepByStudy.getStudySamples({ sample_next_step__sample__id__in: sampleIDs.join(",") }))).data.results
+
+        const studyStepsBySampleID: Record<Sample['id'], FMSSampleNextStepByStudy[] | undefined> = {}
+        for (const sampleNextStepByStudy of sampleNextStepByStudies) {
+            let studySteps = studyStepsBySampleID[sampleNextStepByStudy.sample]
+            if (!studySteps) {
+                studySteps = []
+            }
+            if (!studySteps.find((s) => s.study === sampleNextStepByStudy.study && s.step_order === sampleNextStepByStudy.step_order && s.sample === sampleNextStepByStudy.sample)) {
+                studySteps.push(sampleNextStepByStudy)
+            }
+            studyStepsBySampleID[sampleNextStepByStudy.sample] = studySteps
+        }
+        setStudyStepsBySampleID((old) => ({ ...old, ...studyStepsBySampleID }))
+
+        const studyIDs = new Set<number>()
+        for (const sampleID in studyStepsBySampleID) {
+            for (const studyStep of studyStepsBySampleID[sampleID] ?? []) {
+                studyIDs.add(studyStep.study)
+            }
+        }
+        const studies: FMSStudy[] = []
+        if (studyIDs.size > 0) {
+            studies.push(...await fetchStudies(Array.from(studyIDs)))
+        }
+        const studiesByID: Record<FMSStudy['id'], FMSStudy['letter'] | undefined> = {}
+        for (const study of studies) {
+            studiesByID[study.id] = study.letter
+        }
+        setStudiesByID((old) => ({ ...old, ...studiesByID }) )
+
+        const orderByStepOrder: Record<number, number | undefined> = {}
+        for (const study of studies) {
+            if (study) {
+                const workflow = (await dispatch(api.workflows.get(study.workflow_id))).data
+                for (const stepOrder of workflow.steps_order) {
+                    orderByStepOrder[stepOrder.id] = stepOrder.order
+                }
+            }
+        }
+        setOrderByStepOrder((old) => ({ ...old, ...orderByStepOrder }))
+    }, [dispatch])
+
+    useEffect(() => {
+        if (sampleIDs.length > 0) {
+            refresh(sampleIDs)
+        }
+    }, [refresh, sampleIDs])
+
+    const StudySteps = useCallback(({ sampleID }: { sampleID: Sample['id'] }) => {
+        if (sampleID in studyStepsBySampleID) {
+            return <>{
+                studyStepsBySampleID[sampleID]?.map((studyStep) =>
+                    studiesByID[studyStep.study] && orderByStepOrder[studyStep.step_order]
+                        ? `${studiesByID[studyStep.study]}-${orderByStepOrder[studyStep.step_order]}`
```

Comment:
Would be nice to have a tooltip when mousing over each study step pair that would give the step name.